proposals/lip-state-model.md

```diff
@@ -0,0 +1,103 @@
+This LIP specifies the general format for the key-value storages and the way in which they are combined together in the state tree. 
+
+Organizing the state of a blockchain as a Merkle tree allows to cryptographically authenticate the whole state with a single hash, the state root.
+The state root property is calculated at the end of the block processing and included in the block header.
+When a certificate is posted to a chain, the state root is used to authenticate the information contained in it against the block header, which has been signed by the sending chain validators.
+For example, a node can verify that the cross-chain messages in the certificate payload have been included in the outbox of the sending chain by recomputing the state root signed in the block header from the outbox root.
+This is done using a "witness", i.e. an array of hashes used to recompute the root starting from the known elements of the tree. 
+
+In the future, the state root could further facilitate the introduction of light clients.
+Light clients would only check the validity of block headers, without fully verifying every transaction in a block.
+The state root can then be used to verify the correctness of a portion of the state, such as a user balance, requested by the light client to a full node. 
+
+Furthermore, the state root could be used for faster synchronization: a node could download a snapshot of the blockchain at a block close to the tip, therefore synchronizing quickly to the last state, while in the background it downloads and verifies blocks in the past. 
+In this case, the state root would be used to verify the validity of the snapshot.
+
+Finally, the state root can be employed to improve the [regenesis protocol](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0035.md). 
+A node could download the state root from a trusted party and then check that the genesis block of a chain defines a state that corresponds to the trusted state root.
+
+
+## Rationale
+
+Each module registered in a chain maintains a separate key-value storage. 
+In this LIP, we specify how the entries of the key-value storage are inserted in a sparse Merkle tree (introduced in [LIP 0039][LIP-SMT]), the state tree. 
+The state root is then set to the root of the state tree. 
+The key-value entries of the persistent storage are specified in the respective modules. 
+There is no a-priori separation between user and chain accounts, as they are treated as any generic key-value entry. 
+Each entry specifies a storage prefix and a storage key. 
+The storage prefix identifies different ''buckets'' within the module storage, i.e. group of entries with different scope. 
+The storage key identifies different entries within a bucket.
+
+Keys of the state tree leaf nodes are obtained by concatenating the module ID, the storage prefix and the hash of the storage key. 
```

Comment:
"state tree leaf nodes" sounds odd. You mean "leaf nodes of the state tree"?

Reply:
yes, edited